src/package-request.js

```diff
@@ -151,6 +151,14 @@ export default class PackageRequest {
   }
 
   /**
+   * Remove git protocol prefix in order to match the cache range name.
+   */
+
+  normalizeRegistryRange(range: string): string {
+    return range && range.replace(/^(bitbucket|github|gitlab):/, '');
```

Comment:
I don't think we should remove the provider prefixes since they may be hosting completely unrelated repos. Moreover, even if we decide to do this, I think it should be done by the resolvers themselves so we don't repeat hard-coded protocols around. I also don't think regular expressions are a sustainable way to do this.